### PR TITLE
Convert base GestureHandler class to Kotlin

### PR DESCRIPTION
**Pull request by jakub-gonet**
## Test plan

Tested on Example App.

### CODE REVIEW

File: android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+      }
+      if (hitSlopSet(height)) {
+        if (!hitSlopSet(padTop)) {
+          top = bottom - height
+        } else if (!hitSlopSet(padBottom)) {
+          bottom = top + height
+        }
+      }
+    }
+    return posX in left..right && posY >= top && posY <= bottom
```
Comment (j-piasecki) [inline android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt:367]: Did you have a reason for using range operator for one axis and not for the other, or is it just a mistake?

File: android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+  fun hasCommonPointers(other: GestureHandler<*>): Boolean {
+    for (i in mTrackedPointerIDs.indices) {
+      if (mTrackedPointerIDs[i] != -1 && other.mTrackedPointerIDs[i] != -1) {
+        return true
+      }
+    }
+    return false
+  }
+
+  fun setShouldCancelWhenOutside(shouldCancelWhenOutside: Boolean) = applySelf {
```
Comment (j-piasecki) [inline android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt:85]: For me it was not instantly obvious that this method (and others using `applySelf`) is returning a `ConcreteGestureHandlerT`. Maybe we should state is explicitly?
```suggestion
  fun setShouldCancelWhenOutside(shouldCancelWhenOutside: Boolean): ConcreteGestureHandlerT
    = applySelf {
```

File: android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt (hunk 1 above)
Comment (jakub-gonet) [reply android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt:367]: This is a check for posY being out of range (inclusive).

File: android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt (hunk 2 above)
Comment (jakub-gonet) [reply android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt]: Resolved in b565f2f.

File: android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt (hunk 1 above)
Comment (jakub-gonet) [reply android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt]: Resolved in 3ad3f8c.

**Review by j-piasecki — APPROVED** [review]

**Pull request merged**